src/modules/cpu/kernel/flip.hpp

```diff
@@ -0,0 +1,1148 @@
+    {
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u horizontalFlag = horizontalTensor[batchCount];
+        Rpp32u verticalFlag = verticalTensor[batchCount];
+        
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = (bufferLength / 48) * 48;
+        Rpp32u vectorIncrement = 48;
+        Rpp32u vectorIncrementPerChannel = 16;
+
+        //Initialize hFactor, vFactor and hStrideSrcIncrement with default values (used for getting src pointer location)
+        Rpp32u hFactor = roi.xywhROI.xy.x * layoutParams.bufferMultiplier;
+        Rpp32u vFactor = roi.xywhROI.xy.y * srcDescPtr->strides.hStride;
+        Rpp32s hStrideSrcIncrement = srcDescPtr->strides.hStride;
+
+        //Initialize load functions with default values 
+        auto load48FnPkdPln = &rpp_load48_u8pkd3_to_f32pln3_avx;
+        auto load48FnPlnPln = &rpp_load48_u8pln3_to_f32pln3_avx;
+        auto load16Fn = &rpp_load16_u8_to_f32_avx;
+
+        //Update the load functions, hFactor, vFactor and hStrideSrcIncrement based on the flags enabled
+        if(horizontalFlag == 1)
+        {
+            hFactor += (roi.xywhROI.roiWidth - vectorIncrementPerChannel) * layoutParams.bufferMultiplier;
+            load48FnPkdPln = &rpp_load48_u8pkd3_to_f32pln3_mirror_avx;
+            load48FnPlnPln = &rpp_load48_u8pln3_to_f32pln3_mirror_avx;
+            load16Fn = &rpp_load16_u8_to_f32_mirror_avx;
+        }
+        if(verticalFlag == 1)
+        {
+            vFactor += (roi.xywhROI.roiHeight - 1) * srcDescPtr->strides.hStride;
+            hStrideSrcIncrement = -srcDescPtr->strides.hStride;
+        }
+        srcPtrChannel = srcPtrImage + vFactor + hFactor;
+
+        //Compute constant increment, Decrement factors used in source pointer updation
+        Rpp32s srcPtrIncrement = (horizontalFlag)? -vectorIncrement : vectorIncrement;
+        Rpp32u hFlipFactor = (vectorIncrement - 1) * horizontalFlag;
+        Rpp32s srcPtrIncrementPerChannel = (horizontalFlag)? -vectorIncrementPerChannel : vectorIncrementPerChannel;
+        Rpp32u hFlipFactorPerChannel = (vectorIncrementPerChannel - 1) * horizontalFlag;
+        Rpp32s srcPtrIncrementPerRGB = (horizontalFlag) ? -3 : 3;
+        Rpp32s srcPtrIncrementPerPixel = (horizontalFlag) ? -1 : 1;
+        
+        // flip without fused output-layout toggle (NHWC -> NHWC or NCHW -> NCHW and  horizontalflag = 0 and verticalflag = 0)
+        if ((horizontalFlag == 0) && (verticalFlag == 0) && (srcDescPtr->layout == dstDescPtr->layout))
```

Comment:
please change this to
if (!(horizontalFlag|verticalFlag) && (srcDescPtr->layout == dstDescPtr->layout))

Reply:
done